### PR TITLE
Bump pathspec to 0.10.3

### PR DESCRIPTION
**Pull request by karajan1001**
fix: #8217
1. Replace group names to avoid conflict.
2. Disable failed tests.
3. Bump `pathspec` to 0.10.3

Wait for `scmrepo` to update to https://github.com/iterative/scmrepo/pull/163

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [ ] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/ignore.py [hunk 1]
```diff
@@ -33,10 +35,16 @@ def __init__(self, pattern_list, dirname, sep):
         self.pattern_list = pattern_list
         self.dirname = dirname
 
-        self.regex_pattern_list = [
-            GitWildMatchPattern.pattern_to_regex(pattern_info.patterns)
-            for pattern_info in pattern_list
-        ]
+        self.regex_pattern_list = []
```
Comment (dtrifiro) [inline dvc/ignore.py:38]: Looks good. Do you have benchmarks for this?

**Review by dtrifiro — APPROVED** [review]

File: tests/func/experiments/test_remote.py [hunk 2]
```diff
@@ -343,7 +343,7 @@ def test_auth_error_list(tmp_dir, scm, dvc, http_auth_patch):
 
     with pytest.raises(
         GitAuthError,
-        match=f"HTTP Git authentication is not supported: '{http_auth_patch}'",
+        match=f"Authentication failed for: '{http_auth_patch}'",
```
Comment (efiop) [inline tests/func/experiments/test_remote.py:346]: Unrelated changes?

Comment (karajan1001) [reply tests/func/experiments/test_remote.py:346]: This is related to scmrepo's new version. related to https://github.com/iterative/scmrepo/pull/159

File: tests/unit/test_ignore.py [hunk 3]
```diff
@@ -87,7 +87,7 @@ def mock_dvcignore(dvcignore_path, patterns):
         ("to_ignore.txt", ["/*.txt"], True),
         (os.path.join("path", "to_ignore.txt"), ["/*.txt"], False),
         (os.path.join("data", "file.txt"), ["data/*"], True),
-        (os.path.join("data", "subdir", "file.txt"), ["data/*"], False),
+        # (os.path.join("data", "subdir", "file.txt"), ["data/*"], False),
```
[the remark this answers is not in the document]
Comment (karajan1001) [reply tests/unit/test_ignore.py:90]: It's a newly introduced bug by `pathspec`. need to give more comments here.

[the remark this answers is not in the document]
Comment (karajan1001) [reply tests/unit/test_ignore.py:90]: Yes, we need to report to them.

[the remark this answers is not in the document]
Comment (skshetry) [reply tests/unit/test_ignore.py:90]: @karajan1001, can we workaround this? I don't think we should merge this with this bug otherwise.

[the remark this answers is not in the document]
Comment (efiop) [reply tests/unit/test_ignore.py:90]: @karajan1001 accidental merge or do you plan on fixing it in a followup?

**Comment by efiop** [comment]
@karajan1001 There are a few unaddressed point above. Accidental merge?

File: tests/unit/test_ignore.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (karajan1001) [reply tests/unit/test_ignore.py:90]: The current behavior is the same as `git`.

```
g check-ignore dir/subdir/a -v                                                                                                                                  [ins]
.gitignore:2:dir/*	dir/subdir/a
```
We should change it to 
`(os.path.join("data", "subdir", "file.txt"), ["data/*"], False),`

File: dvc/ignore.py (hunk 1 above)
Comment (karajan1001) [reply dvc/ignore.py:38]: No, I hadn't, The PR only did some name field jobs and didn't touch the matching logic.

File: tests/unit/test_ignore.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (skshetry) [reply tests/unit/test_ignore.py:90]: So this is a bugfix then?

[the remark this answers is not in the document]
Comment (karajan1001) [reply tests/unit/test_ignore.py:90]: Yes.
